lib/parse_stream/ico.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var ParserStream = require('../common').ParserStream;
+
+var HEADER = 0;
+var TYPE_ICO = 1;
+var INDEX_SIZE = 16;
+
+// Format specification:
+// https://en.wikipedia.org/wiki/ICO_(file_format)#Icon_resource_structure
+module.exports = function () {
+  var parser = new ParserStream();
+
+  parser._bytes(6, function (data) {
+    var header = data.readUInt16LE(0);
+    var type = data.readUInt16LE(2);
+    var numImages = data.readUInt16LE(4);
+
+    if (header !== HEADER || type !== TYPE_ICO || !numImages) {
```

Comment:
Here still missed `parser._skipBytes(Infinity);`

When you stop analyse content, the rest should be sent to /dev/null. Or you can block other (parallel) parsers.